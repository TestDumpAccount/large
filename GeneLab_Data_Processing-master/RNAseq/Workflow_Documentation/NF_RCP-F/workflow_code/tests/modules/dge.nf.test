nextflow_process {

    name "Test Process DGE_BY_DESEQ2"
    script "modules/dge.nf"
    process "DGE_BY_DESEQ2"
    
    test("Baseline_ON_GLDS-194:Should run without failures AND PASS VV VALIDATION") {

        when {
            params {
                // define parameters here. Example:
                use_dummy_gene_counts = true
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/GLDS-194/Metadata/GLDS-194_bulkRNASeq_v1_runsheet.csv")
                input[1] = file("test-datasets-extended/testdata/GLDS-194/03-RSEM_Counts/*.genes.results")
                input[2] = [ primary_keytype:'ENSEMBL' ]
                input[3] = file("https://figshare.com/ndownloader/files/36597114")
                input[4] = file("${ baseDir }/bin/dge_annotation_R_scripts.zip")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.dge,
                process.out.norm_counts,
                process.out.dge_ercc,
                process.out.norm_counts_ercc,
                process.out.version,
                ['Passes bulkRNASeq.checks.check_dge_table_group_columns_constraints' : true, 'dp_tools_version' : '1.1.8'] // MANUALLY Validated!
            ).match()
        }

    }

    test("ISSUE_55_ON_GLDS-321:Should run without failures AND PASS VV VALIDATION") {

        when {
            params {
                // define parameters here. Example:
                use_dummy_gene_counts = true
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/GLDS-321/Metadata/GLDS-321_bulkRNASeq_v1_runsheet.csv")
                input[1] = file("test-datasets-extended/testdata/GLDS-321/03-RSEM_Counts/*.genes.results")
                input[2] = [ primary_keytype:'TAIR' ]
                input[3] = file("https://figshare.com/ndownloader/files/36597132")
                input[4] = file("${ baseDir }/bin/dge_annotation_R_scripts.zip")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.dge,
                process.out.norm_counts,
                // NON_ERCC process.out.dge_ercc,
                // NON_ERCC process.out.norm_counts_ercc,
                process.out.version,
                ['Passes bulkRNASeq.checks.check_dge_table_group_columns_constraints' : true, 'dp_tools_version' : '1.1.8'] // MANUALLY Validated!
            ).match()
        }

    }

    test("ISSUE_55_ON_Mock:Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                use_dummy_gene_counts = true
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/mocks/overlapping_samplenames/MOCK_bulkRNASeq_v1_runsheet.csv")
                input[1] = file("test-datasets-extended/testdata/mocks/overlapping_samplenames/*.genes.results")
                input[2] = [ primary_keytype:'ENSEMBL' ]
                input[3] = file("https://figshare.com/ndownloader/files/36597114")
                input[4] = file("${ baseDir }/bin/dge_annotation_R_scripts.zip")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.dge,
                process.out.norm_counts,
                // NON_ERCC process.out.dge_ercc,
                // NON_ERCC process.out.norm_counts_ercc,
                process.out.version,
                ['Passes bulkRNASeq.checks.check_dge_table_group_columns_constraints' : true, 'dp_tools_version' : '1.1.8'] // MANUALLY Validated!
            ).match()
        }

    }

}
