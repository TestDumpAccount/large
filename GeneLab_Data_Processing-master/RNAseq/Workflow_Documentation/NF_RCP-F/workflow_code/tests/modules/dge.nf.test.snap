{
    "Baseline_ON_GLDS-194:Should run without failures AND PASS VV VALIDATION": {
        "content": [
            [
                [
                    "contrasts.csv:md5,66d74b686885ffd4eccdf55823c0e3ce",
                    "SampleTable.csv:md5,bfd18bbc7d34e41c23f0c9107f5d75c9",
                    "differential_expression.csv:md5,00cf45e546529c81c0a43ae1b8495a59",
                    "visualization_output_table.csv:md5,58578caedc33e6a0230ba80abe61f0d5",
                    "visualization_PCA_table.csv:md5,5c461d35b12d5946c2105f705a03c6d3"
                ]
            ],
            [
                [
                    "Normalized_Counts.csv:md5,b4ba348d5446f8ba546a46b966087c1b",
                    "RSEM_Unnormalized_Counts.csv:md5,931c6070b5e19909929c5a217713500b"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                "versions.txt:md5,6e364ecf476a7729d5edd52335fb074a"
            ],
            {
                "Passes bulkRNASeq.checks.check_dge_table_group_columns_constraints": true,
                "dp_tools_version": "1.1.8"
            }
        ],
        "timestamp": "2023-01-25T20:29:00+0000"
    },
    "ISSUE_55_ON_GLDS-321:Should run without failures AND PASS VV VALIDATION": {
        "content": [
            [
                [
                    "contrasts.csv:md5,af3bef64a768dd6220b6a143d2fbb1bc",
                    "SampleTable.csv:md5,0b64b62678b9903bda2a431129cf52af",
                    "differential_expression.csv:md5,e33ffaa350a90f7dd0f4607292db68de",
                    "visualization_output_table.csv:md5,89c4b8722bf2a8fe25c6fcfa915e5c56",
                    "visualization_PCA_table.csv:md5,c19f946356e520bd9bf68606d639f21c"
                ]
            ],
            [
                [
                    "Normalized_Counts.csv:md5,c148732be1d0b1bb61278bfef612f07b",
                    "RSEM_Unnormalized_Counts.csv:md5,fd101e235076c3ae66c513bc96017b33"
                ]
            ],
            [
                "versions.txt:md5,6e364ecf476a7729d5edd52335fb074a"
            ],
            {
                "Passes bulkRNASeq.checks.check_dge_table_group_columns_constraints": true,
                "dp_tools_version": "1.1.8"
            }
        ],
        "timestamp": "2023-01-25T20:29:00+0000"
    },
    "ISSUE_55_ON_Mock:Should run without failures": {
        "content": [
            [
                [
                    "contrasts.csv:md5,b99c88e9c92f1d1588727df47523c4ad",
                    "SampleTable.csv:md5,229c988f09fbfeca182da7011f6f93b4",
                    "differential_expression.csv:md5,f58b6f602598a0c25379afd0c5e87a71",
                    "visualization_output_table.csv:md5,d056472d2ac135cad9ee4d9f33bde387",
                    "visualization_PCA_table.csv:md5,1293b99878d7a7eb0e02dc6a38e33d39"
                ]
            ],
            [
                [
                    "Normalized_Counts.csv:md5,393160aee08165165ccd2b8579a45161",
                    "RSEM_Unnormalized_Counts.csv:md5,6759e0e7ec07960691d3913b3877c129"
                ]
            ],
            [
                "versions.txt:md5,6e364ecf476a7729d5edd52335fb074a"
            ],
            {
                "Passes bulkRNASeq.checks.check_dge_table_group_columns_constraints": true,
                "dp_tools_version": "1.1.8"
            }
        ],
        "timestamp": "2023-01-25T20:29:00+0000"
    }
}